### PR TITLE
Clarify MSI authentication docs for azure native

### PR DESCRIPTION
**Pull request by thomas11**
## Description

This PR expands on some specifics of configuring managed identity authentication for the Azure Native provider.

Resolves pulumi/pulumi-azure-native#3624

### CODE REVIEW

File: themes/default/content/registry/packages/azure-native/installation-configuration.md [hunk 1]
```diff
@@ -297,10 +297,11 @@ as shown above except use `clientCertPath`/`ARM_CLIENT_CERTIFICATE_PATH` instead
...
 
-When MSI is used for authentication, all
-[configuration is done in Azure](https://learn.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview)
-and we only need to tell the provider to use MSI. You can set the Pulumi configuration `useMsi` or the environment
-variable `ARM_USE_MSI` to "true".
+MSI is [configured in Azure](https://learn.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview).
+
+When using a system-assigned identity, or when using a user-assigned identity _and there's only a single one_, you don't need to do anything else. The provider will automatically use the identity.
+
+When one or more resources in your program have multiple user-assigned identities, you need to set the `clientId` config or the `ARM_CLIENT_ID` environment variable to the client ID of the identity you want to use.
```
Comment (mjeffryes) [inline themes/default/content/registry/packages/azure-native/installation-configuration.md:304]: `resources in your program have multiple user-assigned identities` is this something that will be obvious to Azure users? I don't quite follow what it means.

**Review by mjeffryes — APPROVED** [review]
Definitely an improvement. I don't entirely understand what it means for the resources to have multiple identities, but that may just by my ignorance of Azure

File: themes/default/content/registry/packages/azure-native/installation-configuration.md (hunk 1 above)
Comment (thomas11) [reply themes/default/content/registry/packages/azure-native/installation-configuration.md:304]: I _think_ it should be pretty clear to Azure people. It looks like this in a Pulumi program:
```go
			vmIdentity = &compute.VirtualMachineIdentityArgs{
				Type:                   compute.ResourceIdentityTypeUserAssigned,
				UserAssignedIdentities: pulumi.StringArray{umi.ID(), umi2.ID()},
			}
```

Comment (thoward) [reply themes/default/content/registry/packages/azure-native/installation-configuration.md:304]: Clarification on the notion of "user-assigned" vs "system-assigned" identities here: https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/overview#managed-identity-types (maybe we should link to this).

**Pull request merged**